### PR TITLE
feat: support solc output artifact format

### PR DESCRIPTION
**Pull request by mattsse**
## Motivation
Blocked by https://github.com/gakonst/ethers-rs/pull/2024

Closes https://github.com/foundry-rs/foundry/issues/4051

Supports the nested evm object format

### CODE REVIEW

**Comment by mattsse** [comment]
the deny check fails for some unknown reason, also mentioned https://github.com/foundry-rs/foundry/pull/4052#issuecomment-1374469226

File: evm/src/executor/inspector/cheatcodes/ext.rs [hunk 1]
```diff
@@ -682,4 +685,220 @@ mod tests {
         let output = String::decode(&output).unwrap();
         assert_eq!(output, msg);
     }
+
+    #[test]
+    fn test_artifact_parsing() {
+        let s = r#"{
```
Comment (rkrasiuk) [inline evm/src/executor/inspector/cheatcodes/ext.rs:691]: shall we put this in `evm/test-data/`? cautious of bloating source files

Comment (mattsse) [reply evm/src/executor/inspector/cheatcodes/ext.rs:691]: yeh let's do that

**Review by rkrasiuk — APPROVED** [review]
lgtm!

**Pull request merged**